spras/btb.py

```diff
@@ -0,0 +1,163 @@
+
+    #generate input taken from meo.py beacuse they have same input requirements
+    @staticmethod
+    def generate_inputs(data, filename_map):
+        """
+        Access fields from the dataset and write the required input files
+        @param data: dataset
+        @param filename_map: a dict mapping file types in the required_inputs to the filename for that type
+        @return:
+        """
+        for input_type in BowTieBuilder.required_inputs:
+            if input_type not in filename_map:
+                raise ValueError(f"{input_type} filename is missing")
+
+        # Get sources and write to file, repeat for targets
+        # Does not check whether a node is a source and a target
+        for node_type in ['sources', 'targets']:
+            nodes = data.request_node_columns([node_type])
+            if nodes is None:
+                raise ValueError(f'No {node_type} found in the node files')
+
+            # TODO test whether this selection is needed, what values could the column contain that we would want to
+            # include or exclude?
+            nodes = nodes.loc[nodes[node_type]]
+            if node_type == "sources":
+                nodes.to_csv(filename_map["sources"], sep= '\t', index=False, columns=['NODEID'], header=False)
+            elif node_type == "targets":
+                nodes.to_csv(filename_map["targets"], sep= '\t', index=False, columns=['NODEID'], header=False)
+
+
+        # Create network file
+        edges = data.get_interactome()
+
+        # Format into directed graph (BTB uses the nx.DiGraph constructor internally)
+        edges = convert_undirected_to_directed(edges)
+
+        edges.to_csv(filename_map["edges"], sep="\t", index=False,
+                                      columns=["Interactor1", "Interactor2", "Weight"],
+                                      header=False)
+
+
+
+    # Skips parameter validation step
+    @staticmethod
+    def run(sources=None, targets=None, edges=None, output_file=None, container_framework="docker"):
+        """
+        Run BTB with Docker
+        @param sources:  input source file (required)
+        @param targets:  input target file (required)
+        @param edges:    input edge file (required)
+        @param output_file: path to the output pathway file (required)
+        @param container_framework: choose the container runtime framework, currently supports "docker" or "singularity" (optional)
+        """
+
+        # Tests for pytest (docker container also runs this)
+        # Testing out here avoids the trouble that container errors provide
+
+        if not sources or not targets or not edges or not output_file:
+            raise ValueError('Required BowTieBuilder arguments are missing')
+
+        if not Path(sources).exists() or not Path(targets).exists() or not Path(edges).exists():
+            raise ValueError('Missing input file')
+
+        # Testing for btb index errors
```

Comment:
This might not be relevant anymore (if it is, it should be in `prepare_inputs()`). Try commenting out and checking.

Reply:
Did anyone check if this check (for btb index errors) needs to be here, or whether it should be in `prepare_inputs()`? Does it matter?

Reply:
I talked through this with @tristan-f-r - let's note this issue in error handling (that this test isn't checking something that actually will ever happen) and keep the code in for now so that test passes for BTB until we make a decision at a later time.